src/pages/RFOX/components/Shared/SharedMultiStepStatus.tsx

```diff
@@ -0,0 +1,222 @@
+import {
+  CardBody,
+  CardFooter,
+  CardHeader,
+  Center,
+  CircularProgress,
+  CircularProgressLabel,
+  Flex,
+  Heading,
+  Stack,
+} from '@chakra-ui/react'
+import type { Asset } from '@shapeshiftoss/types'
+import { TxStatus } from '@shapeshiftoss/unchained-client'
+import type { PropsWithChildren } from 'react'
+import { useCallback, useMemo, useState } from 'react'
+import { FaCheck } from 'react-icons/fa'
+import { FaX } from 'react-icons/fa6'
+import { useTranslate } from 'react-polyglot'
+import { WithBackButton } from 'components/MultiHopTrade/components/WithBackButton'
+import { SlideTransition } from 'components/SlideTransition'
+import { RawText, Text } from 'components/Text'
+import { fromBaseUnit } from 'lib/math'
+import { selectAssetById } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import type { RfoxBridgeQuote } from '../Bridge/types'
+import { TransactionRow } from './TransactionRow'
+
+type ReusableLpStatusProps = {
+  onBack: () => void
+  confirmedQuote: RfoxBridgeQuote
+} & PropsWithChildren
+
+export const ShareMultiStepStatus: React.FC<ReusableLpStatusProps> = ({
+  confirmedQuote,
+  onBack: handleBack,
+  children,
+}) => {
+  const translate = useTranslate()
+  // TODO(gomes): programmatic, this works for Arbitrum Bridge but we need to find a better way
+  const numSteps = 2
+  const [activeStepIndex, setActiveStepIndex] = useState(0)
+  const [canGoBack, setCanGoBack] = useState(true)
+  const [isFailed, setIsFailed] = useState(false)
+
+  const sellAsset = useAppSelector(state => selectAssetById(state, confirmedQuote.sellAssetId))
+  const buyAsset = useAppSelector(state => selectAssetById(state, confirmedQuote.buyAssetId))
+
+  const bridgeAmountCryptoPrecision = useMemo(
+    () => fromBaseUnit(confirmedQuote.bridgeAmountCryptoBaseUnit, sellAsset?.precision ?? 0),
+    [confirmedQuote.bridgeAmountCryptoBaseUnit, sellAsset?.precision],
+  )
+
+  const steps: { asset: Asset; headerCopy: string; isActionable: boolean }[] = useMemo(() => {
+    if (!(sellAsset && buyAsset)) return []
+    return [
+      {
+        asset: sellAsset,
+        // TODO(gomes): copy
+        headerCopy: `Send ${bridgeAmountCryptoPrecision} ${sellAsset.symbol}`,
+        // TODO(gomes): find a clean way to pass onSignAndBroadcast per step, *if* actionable
+        isActionable: true,
+      },
+      { asset: buyAsset, headerCopy: 'Bridge Funds', isActionable: false },
+    ]
+  }, [bridgeAmountCryptoPrecision, buyAsset, sellAsset])
+
+  // TODO(gomes): handle L2 complete
```

Comment:
Note, with this abstracted, that's now a responsibility of the parent - this will receive L2 Tx and should automagically work